mirascope/core/azure/tool.py

```diff
@@ -87,6 +87,9 @@ def from_tool_call(cls, tool_call: ChatCompletionsToolCall) -> AzureTool:
         Args:
             tool_call: The Azure tool call from which to construct this tool instance.
         """
-        model_json = jiter.from_json(tool_call.function.arguments.encode())
+        if tool_call.function.arguments:
```

Comment:
I think we should update this to the following here and elsewhere:

```python
model_json = {"tool_call": tool_call}
if args := tool_call.function.arguments:
    model_json |= jiter.from_json(args.encode())
return cls.model_validate(model_json)
```

Reply:
note: this will also maintain 100% test coverage so you won't need to make any other changes :)

Reply:
@willbakst I agree with this style recommendation (will work in python 3.9+ and we're on 3.10+)